relay-server/src/actors/outcome.rs

```diff
@@ -475,24 +479,40 @@ pub enum OutcomeError {
 struct HttpOutcomeProducer {
     config: Arc<Config>,
     unsent_outcomes: Vec<TrackRawOutcome>,
-    pending_flush_handle: Option<SpawnHandle>,
+    // Sleep must be `Box<Pin>` in order for it to be Unpin.
+    // See: https://docs.rs/tokio/1.21.0/tokio/time/struct.Sleep.html#examples
+    pending_flush_handle: Either<future::Pending<()>, Pin<Box<tokio::time::Sleep>>>,
 }
 
 impl HttpOutcomeProducer {
     pub fn create(config: Arc<Config>) -> Result<Self, ServerError> {
         Ok(Self {
             config,
             unsent_outcomes: Vec::new(),
-            pending_flush_handle: None,
+            pending_flush_handle: Either::Left(future::pending()),
```

Comment:
So we can totally get rid of `Either` again (and the Pending as it happens), which I'm rather strongly in favour off:
Make `pending_flush_handle` and `Option<Pin<Box<Sleep>>>` and use the precondition in the `select!` call to check for `if pending_flush_handle.is_some()`

Reply:
ugh, this is not true.  the future is still collected, just not polled :(

Reply:
or is it?  the expression is evaluated but maybe it doesn't need to be a future?  would have to try this out to be sure.

Reply:
We tried this out, but this means you have to write this rather ugly as:

```rust
select! {
  () = self.pending_flush_handle.as_mut().unwrap() if self.pending_flush_handle.is_some() => self.do_flush(),
}
```

Compared with this, we preferred using `Either`. However, we also dislike the ambiguity in `Either::Left` and `Either::Right`. This could be solved by introducing a dedicated utility type for this that also has functions like `is_pending()` and `reset()`. Before using `Option` here, I would strongly suggest to write that util.

Reply:
the complexity of `Either` is much higher than the `.is_some()` call and also makes its way into the type system.  `Option` is really well known in comparison.  If the above line works I'd actually prefer it.